dcore/state.py

```diff
@@ -71,24 +77,44 @@ def dump(self):
             self.output.dumplist = self.fieldlist
 
         funcs = self.xn.split()
+        field_dict = {name: func for (name, func) in zip(self.fieldlist, funcs)}
         to_dump = []
-        for name, f in zip(self.fieldlist, funcs):
+        for name, f in field_dict.iteritems():
             if name in self.output.dumplist:
                 to_dump.append(f)
             f.rename(name=name)
 
-        dumpdir = path.join("results", self.output.dirname)
+        if self.output.steady_state_dump_err:
+            init_funcs = self.x_init.split()
+            for name, f, f_init in zip(self.fieldlist, funcs, init_funcs):
+                if name in self.output.dumplist:
+                    err = Function(f.function_space(), name=name+'err').assign(f-f_init)
+                    field_dict[name+"err"] = err
+                    self.diagnostics.register([name+"err"])
```

Comment:
With my change, this just becomes `self.diagnostics.register(name + "err")`.  Which is bit friendlier.